src/jsonapi-datastore.js

```diff
@@ -11,10 +11,47 @@
     this.id = id;
     this._type = type;
     this._attributes = [];
+    this._dependents = [];
     this._relationships = [];
   }
 
   /**
+   * Add a dependent to a model.
+   * @method _addDependence
+   * @param {string} type The type of the dependent model.
+   * @param {string} id The id of the dependent model.
+   * @param {string} key The name of the relation found on the dependent model.
+    */
+  _addDependence(type, id, key) {
+    var self = this,
+      found;
+
+    found = self._dependents.filter(function(dependent) {
+      return dependent.id === id && dependent.type === type;
```

Comment:
Here we should check for relationship key equality as well, as a model may have several back-references from a same model on different relationships (think post has_one author, and post has_many commenters).